realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -995,17 +995,16 @@ boolean contains(Class<? extends RealmObject> clazz) {
      * @param clazz the Class to get objects of.
      * @param fieldName the field name.
      * @return a non-null {@link RealmResults} containing the distinct objects.
-     * @throws IllegalArgumentException if a field name does not exist or the field is not indexed.
+     * @throws IllegalArgumentException if a field name is an empty string.
```

Comment:
All these seems kinda excessive. In other methods we usually only have 1 throws for each type. Perhaps something like "@throws IllegalArgumentException if a field is `null`, doesn't exists, is not indexed or the field points to a linked field."